### PR TITLE
Supporting multiple MS datatypes to be processed on the same instance of SM-platform

### PR DESCRIPTION
**Pull request by iprotsyuk**
- switched to absolute m/z tolerance when computing ion images
- improved of handling LC-MS data
- changed interaction between sm-daemon and RMQ to prevent errors caused by connection timeouts
- supported processing of multiple MS data types on the same server instance
  - includes update of DB schema to distinguish between different ion image storage types: DB and file system

will add changes of tests shortly

### CODE REVIEW

File: sm/engine/dataset_manager.py [hunk 1]
```diff
@@ -170,19 +177,19 @@ def delete(self, ds, del_raw_data=False, **kwargs):
...
 
-    def __init__(self, qname, db, es, mode, queue_publisher=None):
+    def __init__(self, qname, db, es, image_store, mode, queue_publisher=None):
         self.qname = qname
-        DatasetManager.__init__(self, db=db, es=es, mode=mode,
+        DatasetManager.__init__(self, db=db, es=es, img_store=image_store, mode=mode,
                                 queue_publisher=queue_publisher, logger_name='sm-api')
 
     def _post_sm_msg(self, ds, action, priority=DatasetActionPriority.DEFAULT, **kwargs):
+        ds.set_status(self._db, self._es, self._queue, DatasetStatus.QUEUED)
```
Comment (intsco) [inline sm/engine/dataset_manager.py]: @iprotsyuk Why to move it here?

Comment (iprotsyuk) [reply sm/engine/dataset_manager.py]: somehow in VBox installtion a published message is sometimes delivered before the dataset status is set. In this case, setting the status also saves the dataset in DB. So, if a message is processed before that moment, a message handler doesn't find the dataset in DB and the processing fails

File: sm/engine/msm_basic/formula_imager_segm.py [hunk 2]
```diff
@@ -85,8 +88,8 @@ def _gen_iso_images(spectra_it, sp_indexes, sf_peak_df, nrows, ncols, ppm, min_p
 
         # -1, + 1 are needed to extend sf_peak_mz range so that it covers 100% of spectra
         sf_peak_df = sf_peak_df[(sf_peak_df.mz >= sp_df.mz.min()-1) & (sf_peak_df.mz <= sp_df.mz.max()+1)]
-        lower = sf_peak_df.mz.map(lambda mz: mz - mz*ppm*1e-6)
-        upper = sf_peak_df.mz.map(lambda mz: mz + mz*ppm*1e-6)
+        lower = sf_peak_df.mz.map(lambda mz: mz - ABS_MZ_TOLERANCE_DA)
+        upper = sf_peak_df.mz.map(lambda mz: mz + ABS_MZ_TOLERANCE_DA)
```
Comment (intsco) [inline sm/engine/msm_basic/formula_imager_segm.py]: @iprotsyuk this change might cause the sci test to fail. Should be discussed with @andy-d-palmer first

Comment (iprotsyuk) [reply sm/engine/msm_basic/formula_imager_segm.py]: Yeah, I forgot to discuss it before sending the PR. Even if it doesn't break the tests, it's a conceptually important change. The best way to go would be to modify the config to allow for choosing between absolute and relative m/z tolerance

Comment (intsco) [reply sm/engine/msm_basic/formula_imager_segm.py]: Can you choose dynamically based on the datasets type (ims, lcms)?

Comment (iprotsyuk) [reply sm/engine/msm_basic/formula_imager_segm.py]: I can, but one may want to use ppm tolerance for LC-MS data. It's better to make it configurable

Comment (intsco) [reply sm/engine/msm_basic/formula_imager_segm.py]: Then it will require updates on both web app and metadata sides.

File: sm/engine/queue.py [hunk 3]
```diff
@@ -347,32 +347,37 @@ def close_connection(self):
         self.logger.info('Closing connection')
         self._connection.close()
 
-
+# TODO: Currently, a new connection with the queue is created for every queue action.
+# Though it's fine for the actual SM use-cases, where messages are published not frequently,
+# it can be a bottleneck in the future. To overcome this, the code below and annotation job
+# should be rewritten in asyncronous fashion
```
Comment (intsco) [inline sm/engine/queue.py]: @iprotsyuk what was the motivation for creating a new connection for every message to publish?

**Comment by iprotsyuk** [comment]
That was done in order to avoid RabbitMQ connection reset, which led to annotation job failure. Not sure whether it is a glitch on my AWS cluster or it can be reproduced on others. This change is certainly a crutch, but it doesn't affect performance considerably as SM engine doesn't send a big number of messages.

**Comment by intsco** [comment]
@iprotsyuk I merged all the changed into "lcms-support" branch, please take a look and sync

**Pull request merged**